model_compression_toolkit/core/common/graph/memory_graph/memory_graph.py

```diff
@@ -0,0 +1,151 @@
+                memory_tensors.append(memory_tensor)
+                # Add memory tensor as current node's output
+                node_to_tensor.append((n, memory_tensor))
+
+                ot_edges = [oe for oe in out_edges if oe.source_index == i]
+                for oe in ot_edges:
+                    # Add current memory tensor as input to current node's successors
+                    tensor_to_node.append((memory_tensor, oe.sink_node))
+
+        super().__init__(name=model_graph.name + "_memory_graph",
+                         a_nodes=nodes,
+                         b_nodes=memory_tensors,
+                         edges_ab=[nt for nt in node_to_tensor],
+                         edges_ba=[tn for tn in tensor_to_node])
+
+        # memory_lbound_single_op is a lower bound for any schedule of the graph.
+        # the bound is defined as the maximum of memory requirements out of all operations in the graph
+        # (for a single operation the memory requirement is the sum of the memory size of the children and parents)
+        inputs_tensors_memory = [sum([t.total_size for t in self.operation_node_children(n)])
+                                 for n in nodes if n in model_graph.get_inputs()]
+
+        nodes_total_memory = [sum([t.total_size for t in self.operation_node_children(n)] +
+                                  [t.total_size for t in self.operation_node_parents(n)])
+                              for n in nodes if n not in model_graph.get_inputs()]
+
+        self.memory_lbound_single_op = max(nodes_total_memory + inputs_tensors_memory)
+
+        self.sources_a = [n for n in self.a_nodes if len(list(self.predecessors(n))) == 0]
+
+        # Note that unlike the original scheduler,
+        # we don't need sinks_a since we assume all layers have activation tensor.
+        # In oppose to them we do need sinks_b to allow creating single target for astar in case of multiple outputs.
+        self.sinks_b = [n for n in self.b_nodes if len(list(self.successors(n))) == 0]
+        assert len([n for n in self.a_nodes if len(list(self.successors(n))) == 0]) == 0, \
+            "All operations should have an activation tensor, so there are no supposed to be sink nodes in side A," \
+            "of the bipartite memory graph."
+
+    def update_sources_a(self):
+        """
+        Updates the list of Side A source nodes in the bipartite graph.
+        A node is a source if it doesn't have any incoming edges.
+        """
+        self.sources_a = [n for n in self.a_nodes if len(list(self.predecessors(n))) == 0]
+
+    def update_sinks_b(self):
+        """
+        Updates the list of Side B sinks nodes in the bipartite graph.
+        A node is a sink if it doesn't have any outgoing edges.
+        """
+        self.sinks_b = [n for n in self.b_nodes if len(list(self.successors(n))) == 0]
+
+    def activation_tensor_children(self, activation_tensor: ActivationMemoryTensor) -> List[BaseNode]:
```

Comment:
the name is a bit confusing. activation tensor is b_node. maybe replace to "b_node_children". same for a_node. because this is the only place you use the terminology "operation node" and "activation_tensor". what do you think?

Reply:
I want to keep the terminology of "op node" and "activation tensor" because I think it makes the code more understandable. If it would say "b_node" or "a_node" it is not clear what type of object to expect, and then you need to go to the graph definition and check how "a_nodes" and "b_nodes" are defined.
The terms "operation node" and "activation tensor" appear in the MemoryGraph class description, if you think it's not clear enough I can add more comments about it inside the class but I'm not sure if that would help